Backend/src/main/java/com/project/omega/service/implmentations/ProductServiceImpl.java

```diff
@@ -21,10 +25,20 @@
     @Autowired
     ProductRepository productRepository;
 
+    @Autowired
+    SupplierService supplierService;
+
     @Autowired
     MessageSource messages;
 
-    public Product createProduct(Product product) {
+    public Product createProduct(ProductDTO productDTO) throws NoRecordsFoundException {
```

Comment:
Not an emergency but this shouldn't throw a NoRecordsFoundException.

Reply:
`Supplier supplier = supplierService.getSupplierById(productDTO.getSupplierId());`

It's thrown technically by the supplier service, because frontend will be sending only the supplier id, so the supplier needs to be retrieved, and getSupplierById does throw NoRecords.

Reply:
Ah yes of course.